core/src/main/java/tc/oc/pgm/integrations/SimpleVanishIntegration.java

```diff
@@ -84,7 +84,10 @@ public List<MatchPlayer> getOnlineVanished() {
   public boolean setVanished(MatchPlayer player, boolean vanish, boolean quiet) {
     final Match match = player.getMatch();
 
-    // Call vanish event first so name renders existing state properly for leave msg
+    // Set vanish status in match player
+    player.setVanished(vanish);
```

Comment:
Won't this order of events mean vanishing will shows as name crossed in leave message?
I think you need to do the same logic as when calling the event, where it's set to true/false at astart/end lf method depending on if its enabling or disabling

Reply:
@Pablete1234 Do you mean for non-staff broadcasts or staff-only? With the updated logic:

Here is what it looks like when the viewer is a non-staff player, and the staff member runs `/vanish` twice
<img width="303" alt="Screenshot 2023-02-03 at 2 34 40 PM" src="https://user-images.githubusercontent.com/3377659/216723936-0be331b5-c4ce-478f-945b-ddbcf19889ef.png">

Here is what it looks like when the viewer is a staff player, and the other staff member leaves and rejoins the server
<img width="413" alt="Screenshot 2023-02-03 at 2 36 29 PM" src="https://user-images.githubusercontent.com/3377659/216724189-db4b035a-0603-4207-a32f-5665724425c4.png">

So correct in that the name is crossed on the leave message for staff members, but since they're logging out vanished we keep the vanish state right? Personally I don't mind the cross-out on leave messages, helps to further distinguish the player as vanished, but if we did want to remove it we'd probably want to adjust something with the player component? Would that be on the right track or is it something I'm missing?

Reply:
I'm assuming what's happening here is the name renderer for non-staff is rendering the name as "online-and-in-obs" when it should probably be rendering it as offline given vanish status is enabled at the point it's rendering it, which sounds like a different bug.
i don't mind the admin-chat showing crossed names, but the thing is it technically shouldn't, because it should pretend you're leaving while still not vanished (and vanish right as you leave), and so, appear without crossing-out, and then un-vanish right as you join, making the message (again) show as if you were never vanished to the eyes of everyone else

Reply:
> i don't mind the admin-chat showing crossed names, but the thing is it technically shouldn't, because it should pretend you're leaving while still not vanished (and vanish right as you leave), and so, appear without crossing-out, and then un-vanish right as you join

I'm a bit confused about what the solution for this would be 🤔

If I'm understanding it correctly: So upon quit we should always unvanish the player before it gets to the leave message so it appears normally, but still quietly? Though if they're unvanished wouldn't the normal leave broadcast be called? 

My understanding may be totally incorrect, so any guidance on how to make it work the way you described would be helpful. Thanks!

Reply:
> So upon quit we should always unvanish the player before it gets to the leave message so it appears normally, but still quietly?

Upon vanishing, first process the player vanish event and the message (so they "leave" unvanished), and then setVanished to true.

Upon un-vanishing, first setVanished to false, then send the event (so they "join" unvanished).

Optionally you may be able to just get rid of vanished state in match player all together, im not sure what the use of it is atm since the state of vanished or not should come from integration

Reply:
I went ahead and removed the additional vanish state inside of MatchPlayer. It's really unnecessary now that we have the integration, great suggestion. Believe that should solve the issue, thanks again! If there's any other changes you'd like made let me know 👍